mesonbuild/compilers/rust.py

```diff
@@ -323,6 +324,13 @@ def get_rust_tool(self, name: str, env: Environment) -> T.List[str]:
 
         return exelist + args
 
+    def has_multi_arguments(self, args: T.List[str], env: Environment) -> T.Tuple[bool, bool]:
+        return self.compiles('fn main { std::process::exit(0) };\n', env, extra_args=args, mode=CompileCheckMode.COMPILE)
+
+    def has_multi_link_arguments(self, args: T.List[str], env: Environment) -> T.Tuple[bool, bool]:
+        args = self.linker.fatal_warnings() + args
+        return self.compiles(
+            'fn main { std::process::exit(0) };\n', env, extra_args=args, mode=CompileCheckMode.LINK)
```

Comment:
@dcbaker 

Second instance of self.compiles doesn't need to be broken across lines.

If in doubt, prefer to define
```python
code = 'xxxx'
self.compiles(code, env, ....)
```

Reply:
Also in the commit message:
```
Implement both `has_mult_arguments()` and `has_multi_link_arguments()`
to prepare the code accordingly.
```

typo, has_mult(i)_arguments